releasenotes/notes/fix-sparse-z2-sym-11ad38acafb4e5f4.yaml

```diff
@@ -0,0 +1,6 @@
+---
+fixes:
+  - |
+    Fix :meth:`qiskit.quantum_info.analysis.Z2Symmetries.find_z2_symmetries` which was resulting in
+    errors in some cases. The error was due to a mistake in the definition of an escape condition for a
+    loop. This case was not covered by the tests so a corresponding test was added.
```

Comment:
I am not sure whether we need this many "internal details" as part of this release note. That is for the Terra devs to say

Reply:
Was the new z2_symmetries logic even released yet? If not then I would think we would not even need a release note since users would not have seen this bug yet on any stable release.

Reply:
fine by me, I was not aware of that. I'll delete the release note